bionic/core/flow_execution.py

```diff
@@ -192,9 +185,8 @@ def _process_active_entry(self, entry):
                         dst_entry=target_entry,
                         level=EntryLevel.INITIALIZED,
                     )
-            if not entry.all_outgoing_reqs_are_met:
-                self._mark_entry_blocked(entry)
-                return
+                if self._mark_entry_blocked_if_necessary(entry):
```

Comment:
Shouldn't this be outside the for loop (since it's replacing the if condition outside the loop)?

Reply:
Yep! That was a copy-and-paste error. (I guess it still works logically where it is now, but it's more efficient to put it outside the loop.)